### PR TITLE
feat: move logic of dashboard to dashboardApi service

### PR DESCRIPTION
**Pull request by McStOnwelo**
refactor dashboard module with person form component:
1) move request logic to dashboardAPI service
2) clear person form after person edit (bug)

### CODE REVIEW

File: src/app/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -117,16 +101,36 @@ export class DashboardComponent implements OnInit, OnDestroy {
...
       this.isFormVisible = true;
       this.currentForm = formType;
-      if(person) this.editedPerson = person;
+      if(person) {
+        this.editedPerson = person;
+        this.personAdapterService.setEditedPerson(person);
+      }
     }
 
     closeForm(refetch?: boolean | undefined): void {
```
Comment (Bombaharris) [inline src/app/dashboard/dashboard.component.ts]:
```suggestion
    closeForm(refetch = false): void {
```

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -117,16 +101,36 @@ export class DashboardComponent implements OnInit, OnDestroy {
...
+        );
+      }
       this.isFormVisible = false;
       this.currentForm = null;
-      if(refetch) this.queryRef.refetch();
+      this.editedPerson = null;
+      this.personAdapterService.setEditedPerson(null);
+    }
+
+    errorHandler(error: any): void {
```
Comment (Bombaharris) [inline src/app/dashboard/dashboard.component.ts]:
```suggestion
   showError(error: any): void {
```

File: src/app/dashboard/person-form/models/PersonFormModels.ts
Comment (Bombaharris) [inline src/app/dashboard/person-form/models/PersonFormModels.ts:1]: Pleae reneame to ```person.form.model.ts```

File: src/app/dashboard/person-form/person-form.component.ts [hunk 3]
```diff
@@ -1,25 +1,28 @@
...
 export class PersonFormComponent implements OnInit {
-  @Input() person!: Person | any;
+  person!: Person | any;
   @Output() submitted = new EventEmitter();
   @Output() canceled = new EventEmitter();
-  deps$: Observable<DepartmentsQuery['departments']>;
-  projects$: Observable<ProjectsWithAllQuery['projects']>;
-  skills$: Observable<SkillsQuery['skills']>;
-  roles$: Observable<RolesQuery['roles']>;
+  @Output() error = new EventEmitter();
```
Comment (Bombaharris) [inline src/app/dashboard/person-form/person-form.component.ts]:
```suggestion
  @Output() submited: Observalble<Person> = new EventEmitter();
```

File: src/app/dashboard/dashboard.component.ts (hunk 2 above)
Comment (McStOnwelo) [reply src/app/dashboard/dashboard.component.ts]: removed, included in closeForm logic

**Pull request merged**